### PR TITLE
:sparkles: add overlay padding

### LINKED ISSUE #103: Provide padding/margin for the widget overlay
Is there any method to give space between the widget and the overlay? If no, could you please provide it?

Like in this picture (see the red line)

Untitled

### PR DESCRIPTION
**Pull request by vatsaltanna**
Fixes #103

### CODE REVIEW

File: lib/src/get_position.dart [hunk 1]
```diff
@@ -40,10 +43,14 @@ class GetPosition {
         box.size.bottomRight(box.localToGlobal(const Offset(0.0, 0.0)));
 
     final rect = Rect.fromLTRB(
-      topLeft.dx,
-      topLeft.dy,
-      bottomRight.dx,
-      bottomRight.dy,
+      topLeft.dx - padding!.left < 0 ? 0 : topLeft.dx - padding!.left,
```
Comment (BirjuVachhani) [inline lib/src/get_position.dart:46]: Are you sure that the `padding` won't be null here?

File: lib/src/get_position.dart [hunk 2]
```diff
@@ -53,29 +60,29 @@ class GetPosition {
...
-    return bottomRight.dy;
+    return bottomRight.dy + padding!.bottom;
   }
 
   ///Get the top position of the widget
   double getTop() {
     final box = key!.currentContext!.findRenderObject() as RenderBox;
     final topLeft = box.size.topLeft(box.localToGlobal(const Offset(0.0, 0.0)));
-    return topLeft.dy;
+    return topLeft.dy - padding!.top;
```
Comment (BirjuVachhani) [inline lib/src/get_position.dart:70]: Are you sure that the padding won't be null here?

File: lib/src/get_position.dart [hunk 3]
```diff
@@ -53,29 +60,29 @@ class GetPosition {
...
   }
 
   ///Get the left position of the widget
   double getLeft() {
     final box = key!.currentContext!.findRenderObject() as RenderBox;
     final topLeft = box.size.topLeft(box.localToGlobal(const Offset(0.0, 0.0)));
-    return topLeft.dx;
+    return topLeft.dx - padding!.left;
```
Comment (BirjuVachhani) [inline lib/src/get_position.dart:77]: Are you sure that the padding won't be null here?

File: lib/src/get_position.dart [hunk 4]
```diff
@@ -53,29 +60,29 @@ class GetPosition {
...
   }
 
   ///Get the right position of the widget
   double getRight() {
     final box = key!.currentContext!.findRenderObject() as RenderBox;
     final bottomRight =
         box.size.bottomRight(box.localToGlobal(const Offset(0.0, 0.0)));
-    return bottomRight.dx;
+    return bottomRight.dx + padding!.right;
```
Comment (BirjuVachhani) [inline lib/src/get_position.dart:85]: Are you sure that the padding won't be null here?

**Review by BirjuVachhani — CHANGES_REQUESTED** [review]

File: lib/src/get_position.dart (hunk 1 above)
Comment (vatsaltanna) [reply lib/src/get_position.dart]: I have updated the PR. please review the changes.

File: lib/src/get_position.dart (hunk 2 above)
Comment (vatsaltanna) [reply lib/src/get_position.dart]: I have updated the PR. please review the changes.

File: lib/src/get_position.dart (hunk 3 above)
Comment (vatsaltanna) [reply lib/src/get_position.dart]: I have updated the PR. please review the changes.

File: lib/src/get_position.dart (hunk 4 above)
Comment (vatsaltanna) [reply lib/src/get_position.dart]: I have updated the PR. please review the changes.

**Review by BirjuVachhani — APPROVED** [review]
LGTM

**Pull request merged**